### ISSUE TITLE
Add Operating system family property for AWS ECS Fargate Services

### ISSUE THREAD

**Issue by georgikoemdzhiev** [hashicorp/terraform-provider-aws#22033]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Currently the `aws_ecs_service` terraform resources does not have the `platformFamily` property listed on the AWS Service [doc page](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/service_definition_parameters.html)

### New or Affected Resource(s)

- aws_ecs_service

### Potential Terraform Configuration

```
resource "aws_ecs_service" "transform_service" {
  name    = "${var.app_name}-transform-service"
  cluster = aws_ecs_cluster.cluster.id
  task_definition                    = "${aws_ecs_task_definition.transform_service.family}:${max(aws_ecs_task_definition.transform_service.revision, data.aws_ecs_task_definition.tensform_service_td.revision)}"
  desired_count                      = 1
  deployment_minimum_healthy_percent = 50
  deployment_maximum_percent         = 200
  launch_type                        = "FARGATE"
  scheduling_strategy                = "REPLICA"
  runtimePlatform {
    #Valid Values: WINDOWS_SERVER_2019_FULL | WINDOWS_SERVER_2019_CORE | WINDOWS_SERVER_2016_FULL | WINDOWS_SERVER_2004_CORE | WINDOWS_SERVER_2022_CORE | WINDOWS_SERVER_2022_FULL | WINDOWS_SERVER_20H2_CORE | LINUX
    operatingSystemFamily = "WINDOWS_SERVER_2019_CORE"
  
    #Valid Values: X86_64 | ARM64
    cpuArchitecture       = "X86_64"
  }
  network_configuration {
    subnets          = var.subnets
    assign_public_ip = true
    security_groups = [
      aws_security_group.transform_service_sg.id,
    ]
  }
}
```

### References

- https://docs.aws.amazon.com/AmazonECS/latest/developerguide/service_definition_parameters.html
- #21706

**Comment by ChrisMcKee** [comment]
Duplicates #21706

**Comment by kgns** [comment]
I don't think this feature was implemented in #22016

This issue was for `aws_ecs_service` resource, and not for `aws_ecs_task_definition` resource.
Documentation for the latter can be seen on this page: https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_RegisterTaskDefinition.html#ECS-RegisterTaskDefinition-request-runtimePlatform
Documentation for the former can also be seen on this page: https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_Service.html#ECS-Type-Service-platformFamily

We still lack a `platform_family` attribute for `aws_ecs_service` resource. That attribute should force every task registered on that service to be using the same platform family as set for the service.

**Comment by jstehlik-ohpen** [comment]
Update: I've opened a case with AWS Support and experimented with the creating a service via Terraform - according to AWS Support, there is a validation for the `platformFamily` parameter that triggers when you try to create/change the service in the Console, but not in CLI/API. This is consistent with what I've seen during my experiments with creating an ECS service via Terraform - a Windows task was able to successfully start without me explicitly setting `platformFamily`. They promised to update the documentation with more details.

**Comment by ChrisMcKee** [comment]
@jstehlik-ohpen I ran a bunch of fargate windows containers (till I realised just how bloated the pricing is) without issue using the terraform provider. afaik the platform family setting wasn't required by the tasks. 
My original change was written before the cpu type was added and the api docs updated so I'd mapped to platformFamily. The change to the docs meant it was remapped to the runtime; the api doesnt seem to care about the platformFamily tag in the case of fargate then 😕
